content/rs/developing/crdbs/developing-hll-crdb.md

```diff
@@ -0,0 +1,51 @@
+---
+Title: Developing with HyperLogLog in a CRDB
+description:
+weight: $weight
+alwaysopen: false
+categories: ["RS"]
+---
+**HyperLogLog** is an algorithm to address the [count-distinct problem](https://en.wikipedia.org/wiki/Count-distinct_problem).
+To do this it approximates the numbers of items in a [multiset](https://en.wikipedia.org/wiki/Multiset).
+Determining the _exact_ cardinality of a multiset requires memory according to the cardinality of the multiset.
+Because it estimates the cardinality by probability, the HyperLogLog algorithm can run with more reasonable memory requirements.
+
+## HyperLogLog in Redis
+
+Open source Redis implements [HyperLogLog](https://redislabs.com/redis-best-practices/counting/hyperloglog/) (HLL) as a native data-structure.
+It supports adding elements (PFADD) to an HLL, counting elements (PFCOUNT) of HLLs, and merging of (PFMERGE) HLLs.
+
+## HLL in CRDBs versus HLL in Open Source Redis
+
+In CRDBs, we implemented HLL within the CRDT on the basis of the Redis implementation with a few exceptions:
+
+- Redis keeps the HLL data structure as an encoded string object
+    such that you can potentially run any string request can on a key that contains an HLL. In CRDT, only get and set are supported for HLL.
+- In CRDT, if you do SET on a key that contains a value encoded as an HLL, then the value will remain an HLL. If the value is not encoded as HLL, then it will be a register.
+
+## The DEL-Wins approach
+
+Other collections in the Redis-CRDT implementation use the observed remove method to resolve conflicts.
+The CRDT-HLL uses the DEL-wins method.
+If a DEL request is received at the same time as any other request (ADD/MERGE/EXPIRE) on the HLL-key
+the replicas consistently converge to delete key.
```

Comment:
"the CRDB instances consistently converge to delete key."
Don't we use instances in all other pages?